templates/lb-fargate-service/cf.yml

```diff
@@ -1,168 +1,174 @@
       ContainerDefinitions:
-      # TODO: support sidecar images
-        - Essential: true
-          # TODO: support images from different repositories
-          Image: !Sub ${Image}
+        - Name: !Ref AppName
+          Image: !Ref ContainerImage
+          PortMappings:
+            - ContainerPort: !Ref ContainerPort {{if .Variables}}
+          # NOTE Can we magically parametrized this?
+          Environment:{{range $name, $value := .Variables}}
+          - Name: {{$name}}
+            Value: {{$value}}{{end}}{{end}}{{if .Secrets}}
+          Secret:{{range $name, $valueFrom := .Screts}}
+          - Name: {{$name}}
+            ValueFrom: {{$valueFrom}}{{end}}{{end}}
           LogConfiguration:
             LogDriver: awslogs
             Options:
-              awslogs-group: !Ref LogGroup
-              awslogs-stream-prefix: Service
               awslogs-region: !Ref AWS::Region
-          Name: web
-          PortMappings:
-            - ContainerPort: 80
-              Protocol: tcp
-      Cpu: 256
-      ExecutionRoleArn: !GetAtt ExecutionRole.Arn
-      Memory: 512
-      NetworkMode: awsvpc
-      RequiresCompatibilities:
-        - FARGATE
-      TaskRoleArn: !GetAtt TaskRole.Arn
-
-  LogGroup:
-    Type: AWS::Logs::LogGroup
-    UpdateReplacePolicy: Retain
-    DeletionPolicy: Retain
-
+              awslogs-group: !Ref LogGroup
+              awslogs-stream-prefix: ecs
   ExecutionRole:
     Type: AWS::IAM::Role
     Properties:
       AssumeRolePolicyDocument:
         Statement:
-          - Action: sts:AssumeRole
-            Effect: Allow
+          - Effect: Allow
             Principal:
               Service: ecs-tasks.amazonaws.com
-        Version: 2012-10-17
-
-  LogWritingPolicy:
-    Type: AWS::IAM::Policy
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
+  TaskRole:
+    Type: AWS::IAM::Role
     Properties:
-      PolicyDocument:
+      AssumeRolePolicyDocument:
         Statement:
-          - Action:
-              - logs:CreateLogStream
-              - logs:PutLogEvents
-            Effect: Allow
-            Resource: !GetAtt LogGroup.Arn
-        Version: 2012-10-17
-      PolicyName: LogWritingPolicy
-      Roles: [ !Ref ExecutionRole ]
-
+          - Effect: Allow
+            Principal:
+              Service: ecs-tasks.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/AdministratorAccess' # NOTE Are we sure about this??
+  ContainerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: !Join ['', [!Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName, ContainerSecurityGroup]]
+      VpcId:
+        Fn::ImportValue:
+          !Sub "${ProjectName}-${EnvName}-VpcId"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: !Ref ContainerPort
```

Comment:
Changing this config to `IpProtocol: -1` after looking at Nathan's CFN templates: https://github.com/nathanpeck/aws-cloudformation-fargate/blob/master/fargate-networking-stacks/public-private-vpc.yml#L218